server/index.js

```diff
@@ -58,6 +58,126 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    res.json({});
+  } else {
+    const sql = `
+    select "c"."cartItemId",
+      "c"."price",
+      "p"."productId",
+      "p"."image",
+      "p"."name",
+      "p"."shortDescription"
+      from "cartItems" as "c"
+      join "products" as "p" using("productId")
+      where "c"."cartId" = $1
+    `;
+
+    const params = [req.session.cartId];
+
+    db.query(sql, params)
+      .then(result => {
+        const cart = result.rows;
+        res.status(200).json({ cart });
+      });
+  }
+
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = parseInt(req.body.productId, 10);
+  const cartOfPerson = req.session.cartId;
+  if (productId < 0) {
+    res.status(400).json({
+      error: 'productId must be a postive integer'
+    });
+  }
+
+  const sql = `
+    select "price"
+    from "products"
+    where "productId" = $1
+  `;
+
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      const productPrice = result.rows[0];
+      if (!productPrice) {
+        throw res.status(400).json({
+          error: 'No results found'
+        });
+      }
+      if (!req.session.cartId) {
+        const sql = `
+        insert into "carts" ("cartId","createdAt")
+        values(default, default)
+        returning "cartId"
+          `;
+        return db.query(sql)
+          .then(result => result.rows)
+          .then(result => {
+            return {
+              productPrice,
+              result
+            };
+          });
+      }
+      return {
+        productPrice,
+        cartOfPerson
+      };
```

Comment:
This code would be much easier to understand if the result what always an object containing two numbers (cartId and price)